ocaml/xapi/db_gc.ml

```diff
@@ -76,6 +76,53 @@ let gc_VGPU_types ~__context =
 			(String.concat "; " (List.map Ref.string_of (List.map fst garbage)));
 		List.iter (fun (self, _) -> Db.VGPU_type.destroy ~__context ~self) garbage
 
+let gc_PVS_proxies ~__context =
+	(* A proxy connects a VIF, an SR, and a farm. We delete it, only if
+		 all three are invalidi. XXX Is it correct to consider cache_SR?  *)
+	let destroy self =
+		( debug "GC-ing PVS_proxy %s" (Ref.string_of self)
+		; Db.PVS_proxy.destroy ~__context ~self
+		) in
+	let is_alive proxy =
+		(**) valid_ref __context @@ Db.PVS_proxy.get_VIF      ~__context ~self:proxy
+		&&   valid_ref __context @@ Db.PVS_proxy.get_farm     ~__context ~self:proxy
+		&&   valid_ref __context @@ Db.PVS_proxy.get_cache_SR ~__context ~self:proxy in
+	Db.PVS_proxy.get_all ~__context
+	|> List.filter (fun proxy -> not @@ is_alive proxy)
+	|> List.iter destroy
+
+let gc_PVS_servers ~__context =
+	(* A PVS server refers to a PVS farm. We delete it, if the reference
+	 * is invalid *)
+	let destroy self =
+		( debug "GC-ing PVS_server %s" (Ref.string_of self)
+		; Db.PVS_server.destroy ~__context ~self
+		) in
+	let is_alive server =
+		valid_ref __context @@ Db.PVS_server.get_farm ~__context ~self:server in
+	Db.PVS_server.get_all ~__context
+	|> List.filter (fun server -> not @@ is_alive server)
+	|> List.iter destroy
+
+let gc_PVS_farms ~__context =
```

Comment:
I don't think we want to GC farms - this will mean that a new farm, once created, runs the risk of getting destroyed before the user creates any servers or proxies in it.